### ISSUE TITLE
在 SecretFlow 中实现针对拆分学习的 sota label 攻击和防御方法

### ISSUE THREAD

**Issue by Candicepan** [secretflow/secretflow#745]
> 此 ISSUE 为 [隐语开源共建计划（SecretFlow Open Source Contribution Plan，简称 SF OSCP）](https://github.com/orgs/secretflow/discussions/1181)任务 ISSUE，欢迎社区开发者参与共建～
> 若有感兴趣想要认领的任务，但还未报名，辛苦先完成[报名](https://www.wjx.top/vm/QKio0dq.aspx#)进行哈～

## 任务介绍

- 任务名称：在 SecretFlow 中实现针对拆分学习的 sota label 攻击和防御方法
- 技术方向：联邦学习和人工智能
- 任务难度：进阶🌟🌟
- **任务期望时间**：4-5周

## 详细要求

- 功能性：在 SecretFlow 中实现以下功能代码以及实现过程文档，功能包括：
  - 需要引入 label 攻防算法相关的数据集，并提供拆分逻辑
  - 基于 SLModel 实现 label 攻防代码，攻防算法可以通过 SLModel 中 callback 实现，或另外写 label 攻击防御算法类实现
  - 需要同时提交 tutorial，example 代码，数据下载，拆分和预处理代码
- 收敛性/准确性：包含 simulator 跑出的实验数据并且给出攻击防御指标，最好有与明文结果（PyTorch 或者 TensorFlow）的对比
- 代码规范：Python 代码需要使用 black+isort 进行格式化（流水线包含代码规范检查卡点）
- 提交说明：
  - 关联该 issue 并提交代码（位置待补充）
  - 整理完整实现过程为文档（位置待补充）
 
## 能力要求
- 熟悉拆分学习攻防方法
- 熟悉 TensorFlow 或者 PyTorch

## 操作说明
- 参考内容：[联邦模型开发教程](https://www.secretflow.org.cn/docs/secretflow/latest/en-US/tutorial/CustomDataLoaderOnSL)
